frontend/src/pages/PasswordResetRequest.tsx

```
@@ -0,0 +1,105 @@
+import React, { useState } from "react";
+
+type Lang = "uk" | "en";
+type Props = { lang?: Lang };
+
+const TR = {
+  uk: {
+    title: "Відновлення пароля",
+    emailLabel: "Електронна пошта",
+    submit: "Надіслати запит",
+    required: "Поле електронної пошти є обов’язковим.",
+    invalid: "Введіть коректну адресу електронної пошти.",
+    success: "Ми надіслали інструкції з відновлення пароля на вашу електронну пошту.",
+    throttled: "Забагато запитів. Спробуйте пізніше.",
+  },
+  en: {
+    title: "Password recovery",
+    emailLabel: "Email",
+    submit: "Send request",
+    required: "Email is required.",
+    invalid: "Enter a valid email address.",
+    success: "We have sent password reset instructions to your email address.",
+    throttled: "Too many requests. Please try again later.",
+  },
+} as const;
+
+const EMAIL_RE = /\S+@\S+\.\S+/;
+
+export default function PasswordResetRequest({ lang = "uk" }: Props) {
+  const t = TR[lang];
+  const [email, setEmail] = useState("");
+  const [error, setError] = useState<string | null>(null);
+  const [msg, setMsg] = useState<string | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  async function onSubmit(e: React.FormEvent) {
+    e.preventDefault();
+    setError(null);
+    setMsg(null);
+
+    const v = email.trim();
+    if (!v) return setError(t.required);
+    if (!EMAIL_RE.test(v)) return setError(t.invalid);
+
+    setLoading(true);
+    try {
+      const r = await fetch("/api/auth/password-reset/", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({ email: v }),
+      });
+      setMsg(r.status === 429 ? t.throttled : t.success);
```

The component intentionally shows the same "success" message for 200, 400, 500, and even network errors. While preventing account enumeration is valid, fully swallowing server errors (500) or logging nothing makes real service issues invisible to users and maintainers.

Keep the non‑enumeration behavior, but differentiate internal failures for debugging/ops: e.g., show the generic "If an account exists..." message to the user in all cases, but log the server error to a telemetry/logging system and show a transient "Try again later" for 5xx or network failures if appropriate. Minimal code change:
```
const r = await fetch(...);
if (r.status === 429) setMsg(t.throttled);
else setMsg(t.success);
if (!r.ok && r.status >= 500) {
  // send to logging/monitoring
  console.error("password-reset server error", r.status);
  // optionally: set an unobtrusive "try again later" in addition to generic msg
}
```

---

done